src/sprints/sprints.controller.spec.ts

```diff
@@ -1,25 +1,1261 @@
-import { Test, TestingModule } from "@nestjs/testing";
-import { SprintsController } from "./sprints.controller";
-import { SprintsService } from "./sprints.service";
-
-describe("SprintsController", () => {
-    let controller: SprintsController;
-
-    beforeEach(async () => {
-        const module: TestingModule = await Test.createTestingModule({
-            controllers: [SprintsController],
-            providers: [
-                {
-                    provide: SprintsService,
-                    useValue: {},
+import { Test, TestingModule } from '@nestjs/testing';
+import { SprintsController } from './sprints.controller';
+import { SprintsService } from './sprints.service';
+import { CustomRequest } from '@/global/types/CustomRequest';
+import { createMockData, mockDate } from './mock-data';
+import { CreateTeamMeetingDto } from './dto/create-team-meeting.dto';
+import { UpdateTeamMeetingDto } from './dto/update-team-meeting.dto';
+import { CreateAgendaDto } from './dto/create-agenda.dto';
+import { UpdateAgendaDto } from './dto/update-agenda.dto';
+import { UpdateMeetingFormResponseDto } from './dto/update-meeting-form-response.dto';
+import { CreateCheckinFormDto } from './dto/create-checkin-form.dto';
+import {
+  AgendaResponse,
+  CheckinFormResponse,
+  CheckinSubmissionResponse,
+  MeetingFormResponse,
+  MeetingResponse,
+} from './sprints.response';
+import { CheckinQueryDto } from './dto/get-checkin-form-response';
+
+describe('SprintsController', () => {
+  let sprintsController: SprintsController;
+  let sprintsService: SprintsService;
+
+  const mockRequest = createMockData.request as unknown as CustomRequest;
+
+  const mockFormResponseMeeting: MeetingFormResponse = {
+    id: 1,
+    meetingId: 1,
+    formId: 1,
+    responseGroupId: 1,
+    createdAt: mockDate,
+    updatedAt: mockDate
+  };
+  const mockFormWithQuestionsResponse = createMockData.formWithQuestions(1, mockFormResponseMeeting);
+
+  const mockCreateMeetingDto: CreateTeamMeetingDto = {
+    title: "Sprint Planning",
+    description: "Planning session for sprint 1",
+    meetingLink: "http://meet.google.com/123",
+    dateTime: mockDate,
+    notes: "Please come prepared",
+  };
+
+  const mockUpdatedMeetingDto: UpdateTeamMeetingDto = {
+    title: "Updated Sprint Meeting",
+    description: "Updated Planning session for sprint 1" ?? null,
+    meetingLink: "http://meet.google.com/123Updated" ?? null,
+    dateTime: mockDate ?? null,
+    notes: "Updated Please come prepared" ?? null,
+  };
```

Comment:
I think we don't  need  to use  `??` operator  as the left hand side will never be null